### PR TITLE
EQPS-908: Devdocs for EQP MFTF Tests

### PR DESCRIPTION
**Pull request by magterskine**
## Purpose of this pull request

This pull request (PR) adds pages for new EQP test types to Marketplace EQP Sellers pages. These Types are MFTF Vendor Supplied tests and MFTF Magento Tests

## Affected DevDocs pages

- Adds new pages here: https://devdocs.magento.com/marketplace/sellers/...

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
@magterskine thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by BarnyShergold** [comment]
@magterskine - Your two files added are blank, so I'm assuming this is still in progress? As you have not put the PR in Draft mode, can you change the title to prefix with [WIP] and then remove when ready to review? Then the maintainers know when the PR is ready for review. Thanks.

**Comment by magterskine** [comment]
> @magterskine - Your two files added are blank, so I'm assuming this is still in progress? As you have not put the PR in Draft mode, can you change the title to prefix with [WIP] and then remove when ready to review? Then the maintainers know when the PR is ready for review. Thanks.

@BarnyShergold - Apologies, content wasnt synced. New commits added to populate both new files with content. Now ready for review.

**Comment by dobooth** [comment]
Hi @magterskine Can you please enable maintainer editability please. https://docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/giving-team-maintainer-permissions-to-an-organization-member

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:9]:
```suggestion
The Magento Functional Testing Framework vendor supplied Tests are a browser-based user acceptance testing framework used to validate the functionality of an operating Magento site. Running Vendor-Supplied MFTF tests helps ensure that the extension functionality is operating as expected for the end user.
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:18]:
```suggestion
MFTF runs tests in a browser, using Selenium and Codeception, to emulate user behavior. MFTF tests are designed to be extensible and be used in conjunction with Magento developed MFTF code coverage. This allows vendor supplied, extension MFTF tests to describe the behavior of the extension and to integrate with existing suites of MFTF coverage, leveraging reusable tests elements and modifying existing test flows as necessary.
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:5]: Remove extra empty line

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:12]: Remove extra empty line

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:21]: Remove extra empty line

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:27]:
```suggestion
-  There are MFTF tests included in the submission in the correct directory (`Tests/Mftf`)
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:30]: Remove extra empty line

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:34]:
```suggestion
Vendor supplied MFTF tests will be run on an installed Magento instance of the type listed above. These tests will help verify that the functionality intended is testable and working on a Magento environment. The results of this testing can be used to inform Marketplace QA resources about the state of the extension and what level of effort will be required in manually verifying the extension. A higher level of vendor supplied MFTF coverage will allow manual QA efforts to be focussed and potentially accelerate the progression of the extension through the Technical Review process.
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:40]: Remove extra empty lines

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:45]: Numbered lists all have 1. for all items
```suggestion
1. Allure results (as XML)
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:43]:
```suggestion
MFTF returns two types of results:
```
Add extra line after and before list

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:47]:
```suggestion
See [MFTF Reporting](https://devdocs.magento.com/mftf/docs/reporting.html) for further information.
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:53]: Remove extra empty line

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:49]:
```suggestion
To diagnose a test failure, you should first ascertain if you can replicate the failure on your local development environment. If you cannot, this indicates a configuration and/or environment difference.
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:55]: Remove empty lines between each item in this list.

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:61]:
```suggestion
-  MFTF within this directory should follow the standard directory structure, separating ActionGroups, Tests, Pages, Sections etc, into their own directories
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:63]:
```suggestion
-  Given that tests will be executed with Magento MFTF tests available for merging, please ensure that they will operate as expected in this configuration, e.g. Ensure that test names do not clash with existing Magento tests
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:65]:
```suggestion
-  Ensure that if you are extending from, or merging into, an existing Magento test (or relying upon its entities), that it is required as a composer prerequisite
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:69]: Avoid contractions
```suggestion
-  Do not include any sensitive or confidential data in test comments or code
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:71]:
```suggestion
-  If necessary, include a README, or other instructions, within the `Test` directory, to explain any setup steps or caveats on running your MFTF tests
```

File: src/marketplace/sellers/mftf-vendor.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-vendor.md:5]: Remove extra empty line

File: src/marketplace/sellers/mftf-vendor.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-vendor.md:1]: As this file is identical to the first file, please apply all corrections show above to this file.

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Great new feature = mainly language & punctuation corrections.

**Comment by magterskine** [comment]
PR updated - different content for MFTF Vendor and MFTF Magento pages.
Changes as requested made.
Linting errors fixed.

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:8]:
```suggestion
The Magento Functional Testing Framework Magento supplied Tests are a browser-based user acceptance testing framework used to validate the functionality of an operating Magento site. Running Magento supplied MFTF tests aids EQP QA efforts in assessing critical functionality of a Magento instance with the extension installed.
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:38]:
```suggestion
   -  See [MFTF Reporting](https://devdocs.magento.com/mftf/docs/reporting.html) for further information
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:40]:
```suggestion
   -  MFTF test flow being broken due to changes to Magento user flows as intended by the extension under test. These can be fixed by using vendor supplied MFTF Tests.
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:41]:
```suggestion
   -  A failure in Magento user flows due to an unintended side-effect of the extension under test. This may be a type of regression bug and will be tested and verified by manual QA as part of EQP technical review.
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:51]:
```suggestion
-  Ensure that if you are extending from, or merging into, an existing Magento test (or relying upon it's entities), that it is required as a composer prerequisite
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:53]:
```suggestion
-  Do not include any sensitive or confidential data in test comments or code
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [reply src/marketplace/sellers/mftf-magento.md:5]: This hasn't been fixed

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [reply src/marketplace/sellers/mftf-magento.md:49]: This hasn't been fixed

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [reply src/marketplace/sellers/mftf-magento.md:53]: This hasn't been fixed

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Found some additional corrections, plus some items not fixed. Please address and apply to second file.

**Comment by BarnyShergold** [comment]
@magterskine - Please can you fix the outstanding issues on this PR?

**Comment by magterskine** [comment]
@BarnyShergold 
Changes made as requested.
Apologies for delay, significant disruption in Texas due to storm.

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:50]:
```suggestion
-  Given that tests will be executed with Magento MFTF tests available for merging etc, please ensure that they will operate as expected in this configuration e.g. Ensure that test names do not clash with existing Magento tests
```

File: src/marketplace/sellers/mftf-magento.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-magento.md:51]:
```suggestion
-  Ensure that if you are extending from, or merging into, an existing Magento test (or relying upon its entities), that it is required as a composer prerequisite
```

File: src/marketplace/sellers/mftf-vendor.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-vendor.md:14]:
```suggestion
To ensure that extension submissions perform the expected behavior for a user, MFTF tests are used to describe and confirm the functionality expected via browser-based testing against an operational Magento instance.
```

File: src/marketplace/sellers/mftf-vendor.md
Comment (BarnyShergold) [inline src/marketplace/sellers/mftf-vendor.md:59]:
```suggestion
-  Ensure that any necessary credentials, user authorization (e.g. to communicate via API key to your backend) is explained and supported via credential management in tests
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
I've been through and found 4 very small changes - if you can address these then I will approve.

**Comment by magterskine** [comment]
Thanks for the quick turnaround. Changes made.

**Review by BarnyShergold — APPROVED** [review]
Nice work!

**Comment by keharper** [comment]
@magterskine 

Since @dobooth is the MFTF guru and because he made one comment already, I was going to make a drive-by comment and mention that the TOC file needs to be updated so that the new topics show up in the left navigation. I looked at the Marketplace Sellers guide and read the content of this PR, and ... I think we need to reconsider how this information should be presented.

My recommendations:

* I think the general discussion of MFTF should be in "Technical Review Guidelines". This will reduce the duplicated overview text.
* Combine the two current topics into one topic, with one Head2 devoted to Magento-supplied testing, and another devoted to vendor-supplied testing. The topic should probably be placed near "Installation & Varnish Tests" but I defer to you for exact placement.
* We're missing details about submitting MFTF results. Most likely, the EQP API docs need to be updated to show how to include the results. If that information is already there, then a link to correct REST endpoint would suffice.
* You'll probably need to make passing references to MFTF in places like "Create an Extension" and "Submit for Technical Review".

If I were doing a copy edit right now, I'd be leaving a lot of comments. It would be easier if you would enable maintainer editability, as Don previously requested.

Let Don and I know if you want to set up a meeting.

**Comment by vkublytskyi** [comment]
@keharper I do agree that Marketplace Sellers structure should be reviewed but let's try to minimize scope of this PR and address all needed changes in consequence PRs. Our team would be glad to hear and implement any recommendations and best practices for our documentation.

From our perspective, at this moment we do really need 2 separate pages as they represent 2 separate checks that we have in EQP. For later this year we have an epic to combine these checks. Once it will be done we will update documentation and replace these pages with a single. Also, we have a links from Developer Portal UI to documentation so sellers may easily find all information for a particular check so for this we would prefer some text duplication on several pages to simplify information navigation for vendors.

@magterskine Please add links to new pages in `Submit for Technical Review` under `Installation and Varnish Test`. You may find similar changes at https://github.com/magento/devdocs/pull/8112/files#diff-aa0ea4b2b4fcf21a6e890b9f7bbe4b08bd3cec6b36fbb2f9105b4afa58ed9fff. Also, as mentioned by @keharper our [Technical Review Guidelines](https://devdocs.magento.com/marketplace/sellers/technical-review-guidelines.html) should have a short overview of MFTF tests with links to the main pages (also below `Installation and Varnish Tests: Verify that product installs and caching works correctly`).

**Comment by magterskine** [comment]
Made changes as Volodymyr's request.

@keharper - I am not able enable the maintainability -I'll reach out to you on Slack to see if you can point me in the right direction.

**Comment by dobooth** [comment]
Moving this to another branch so the writing team can make edits.

**Comment by keharper** [comment]
@magterskine @vkublytskyi  I'm going to let @dobooth provide most of the commentary from here on out. It seems that a discussion about uploading the test results is needed, perhaps in the EQP API reference. It's OK if that issue and the documentation restructuring occur in subsequent PRs.

**Pull request merged**
